lib/src/text_view.dart

```diff
@@ -372,27 +378,47 @@ class _RichTextViewState extends State<RichTextView> {
         textPainter.layout(minWidth: constraints.minWidth, maxWidth: maxWidth);
         final ellipsisSize = textPainter.size;
 
+        // First measure content without the prefix to avoid WidgetSpan dimension issues
         textPainter.text = content;
         textPainter.layout(minWidth: constraints.minWidth, maxWidth: maxWidth);
-        final textSize = textPainter.size;
+        final contentSize = textPainter.size;
+        final contentExceedsMaxLines = textPainter.didExceedMaxLines;
+
+        // Estimate prefix width instead of directly measuring the WidgetSpan
+        final prefixWidth = widget.prefixIconWidth ?? 0;
+
+        // Determine if text will exceed max lines with the prefix
+        final exceedsMaxLines = contentExceedsMaxLines ||
```

Comment:
Instead, let's try to modify the `content` TextSpan to include the prefix widget span, that should be much simpler and there should be no need to use this custom exceedsMaxLines logic:

![image](https://github.com/user-attachments/assets/504cf9ec-1191-424e-adea-b112e812b564)